python/cudf/cudf/core/column/categorical.py

```diff
@@ -936,6 +946,22 @@ def unary_operator(self, unaryop: str):
         )
 
     def __setitem__(self, key, value):
+        if cudf.utils.dtypes.is_scalar(value):
+            new_values = [value]
```

Comment:
Do we need this scalar check? It seems like we can just construct the index down below by passing the scalar directly. 
```
>>> cudf.Index(2)
Int64Index([2], dtype='int64')
```

Reply:
Didn't know we already handle it in Index constructor, removed the check.